contrib/meson/meson/examples/meson.build

```diff
@@ -26,7 +26,7 @@ foreach e, src : examples
   executable(
     e,
     lz4_source_root / 'examples' / src,
-    dependencies: liblz4_dep,
+    dependencies: [liblz4_dep, liblz4_internal_dep],
```

Comment:
What is `liblz4_internal_dep` ?

Reply:
I removed `liblz4_dep` as `liblz4_internal_dep` is necessary for static linkage of lz4. Having both is redundant. 
It's defined [here](https://github.com/lz4/lz4/blob/dev/contrib/meson/meson/lib/meson.build#L47) as `static_library(...)` for example `contrib/meson/meson/programs/meson.build` is using the static linkage as well.